### PR TITLE
Issue 36912: Allow admins to dismiss "static" warnings

### PR DESCRIPTION
**Pull request by labkey-adam**

### CODE REVIEW

File: core/src/org/labkey/core/view/template/bootstrap/WarningServiceImpl.java [hunk 1]
```diff
@@ -37,4 +51,100 @@ public void forEachProvider(Consumer<WarningProvider> consumer)
...
+        if (ModuleLoader.getInstance().isStartupComplete())
+        {
+            // We should have our full list of warnings at this point, so safe to cache them
+            STATIC_ADMIN_WARNINGS = messages;
+        }
+        return messages;
+    }
+
+    private static final String DISMISSAL_SCRIPT_FORMAT =
+        "<script type=\"text/javascript\">\n" +
```
Comment (labkey-jeckels) [inline core/src/org/labkey/core/view/template/bootstrap/WarningServiceImpl.java]: Should we be somewhere declaring a dependency on jQuery?

File: core/src/org/labkey/core/view/template/bootstrap/WarningServiceImpl.java [hunk 2]
```diff
@@ -37,4 +51,100 @@ public void forEachProvider(Consumer<WarningProvider> consumer)
...
+        {
+            List<String> messages = warnings.getMessages();
+
+            if (messages.size() == 1)
+                html.append(messages.get(0));
+            else
+            {
+                html.append("<ul>");
+                for (String msg : messages)
+                    html.append("<li>").append(msg).append("</li>");
```
Comment (labkey-jeckels) [inline core/src/org/labkey/core/view/template/bootstrap/WarningServiceImpl.java]: Is this known to be safe HTML to embed? This isn't an HTMLString

Comment (labkey-adam) [reply core/src/org/labkey/core/view/template/bootstrap/WarningServiceImpl.java]: I guess if we're going to the bother of changing the interface we might as well force all the messages to HtmlString. That expands the footprint into LDK, etc., but is probably good to do now.

File: core/src/org/labkey/core/view/template/bootstrap/WarningServiceImpl.java (hunk 1 above)
Comment (labkey-adam) [reply core/src/org/labkey/core/view/template/bootstrap/WarningServiceImpl.java]: It's on every page, see `PageFlowUtil.getIncludes()`

**Comment by labkey-adam** [comment]
Also a change to LDK in SVN... because of the natural sort CLR installation manager. Same change as group_concat and premium stats.

**Pull request merged**